models/eprop_iaf_adapt.h

```diff
@@ -0,0 +1,662 @@
+/*
+ *  eprop_iaf_adapt.h
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef EPROP_IAF_ADAPT_H
+#define EPROP_IAF_ADAPT_H
+
+// nestkernel
+#include "connection.h"
+#include "eprop_archiving_node_impl.h"
+#include "eprop_archiving_node_recurrent.h"
+#include "eprop_synapse.h"
+#include "event.h"
+#include "nest_types.h"
+#include "ring_buffer.h"
+#include "universal_data_logger.h"
+
+namespace nest
+{
+
+/* BeginUserDocs: neuron, e-prop plasticity, current-based, integrate-and-fire, adaptive threshold
+
+Short description
++++++++++++++++++
+
+Current-based leaky integrate-and-fire neuron model with delta-shaped
```

Comment:
The comment here seems wrong: This model does not implement delta-synapses (they are in eprop_iaf_psc_delta), but instead has synaptic input via the z variable, which is 1 for one time step, if I am not wrong?